metadata_store_scripts/V20205.01.30.0__create_constraints_stored_procedure.sql

```diff
@@ -0,0 +1,147 @@
+
+-- Create a table type to store constraint information
+-- before dropping the constraints
+CREATE TYPE sink_drop_constraint_type AS TABLE
+(
+    pipeline_run_id NVARCHAR(255),
+    dataset NVARCHAR(255),
+    specified_database NVARCHAR(255),
+    specified_schema NVARCHAR(255),
+    identified_parent_table NVARCHAR(255),
+    child_table NVARCHAR(255),
+    constraint_name NVARCHAR(255),
+    children_columns NVARCHAR(MAX),
+    parent_columns NVARCHAR(MAX),
+    pre_drop_status NVARCHAR(255),
+    drop_timestamp DATETIME,
+    post_create_status NVARCHAR(255)
+);
+
+-- Create a stored procedure to insert constraint information
+-- into the capture_constraints table
+CREATE PROCEDURE insert_constraints
+    @source sink_drop_constraint_type READONLY
+AS
+BEGIN
+    MERGE [dbo].[capture_constraints] AS ct
```

Comment:
There should be no references to `dbo` in these migrations as customers should have an easy way of using a different schema if they so choose, and hard-coding the `dbo` makes that more difficult. For our tests, usually this doesn't matter since `dbo` is the default schema, but that default schema can be changed on a per-user basis: https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-user-transact-sql?view=sql-server-ver16

Note: I forgot this in my last PR, so I'll open another PR to correct them